### ISSUE TITLE
Improvements needed in pdfreport tool

### ISSUE THREAD

**Issue by asaunier** [camptocamp/c2cgeoportal#1236]
* ~~Doc (http://docs.camptocamp.net/c2cgeoportal/1.5/integrator/pdfreport.html) => add doc for missing .jar files (including postgresql JDBC driver). Unless they are added in MF Print v3? See https://github.com/camptocamp/schwyz_geoportal/commit/ac9db68e6367eaaf36ebc3f94e1681090c28beaf~~

* ~~If no map is embedded into the report, skip the WFS request URL generation https://github.com/camptocamp/c2cgeoportal/blob/1.5/c2cgeoportal/views/pdfreport.py#L91-L107 and do not complain if SRS is not set in the config~~

* ~~The "id" argument could also be a string, not only an integer: https://github.com/camptocamp/c2cgeoportal/blob/1.5/c2cgeoportal/__init__.py#L294~~

**Comment by asaunier** [comment]
> Doc (http://docs.camptocamp.net/c2cgeoportal/1.5/integrator/pdfreport.html) => add doc for missing .jar files (including postgresql JDBC driver). Unless they are added in MF Print v3? See camptocamp/schwyz_geoportal@ac9db68

@sbrunner Could you remind me where those JAR files (with the postgres JDBC driver) are supposed to be downloaded? I would like to update the doc and add the links. Thanks!

**Comment by tsauerwein** [comment]
jasperreports-fonts is now included in mfp (version >= 3.1.0), see https://github.com/mapfish/mapfish-print/pull/187

**Comment by asaunier** [comment]
> jasperreports-fonts is now included in mfp (version >= 3.1.0), see mapfish/mapfish-print#187

Thanks Tobias, it indeed works without adding this file. What about including the others in mfp?

**Comment by tsauerwein** [comment]
> What about including the others in mfp?

postgresql-9.3-1102.jdbc41.jar
-> Rather not, because this is project specific.

joda-time-1.6.jar
-> Might be useful, but I also tend to say that this is project specific.

jasperreports-functions-5.5.0.jar
-> Why not. Do you want to make a PR?

**Comment by asaunier** [comment]
> jasperreports-functions-5.5.0.jar
-> Why not. Do you want to make a PR?

Well, I guess I have to add something after https://github.com/mapfish/mapfish-print/blob/development/core/build.gradle#L97
most likely
```
'net.sf.jasperreports:jasperreports-functions:5.5.0'
```
but I am not sure and don't really nknow how to test the change :)

**Comment by tsauerwein** [comment]
You could create a WAR with `./gradlew war` and then check if the WAR contains the `jasperreports-functions` JAR file. But I can also do it if you want to.

**Comment by asaunier** [comment]
> But I can also do it if you want to.

This option looks good to me :P

**Comment by tsauerwein** [comment]
> This option looks good to me :P

Hehe, in fact it's not that easy. :) First, `jasperreports-functions` in not in an official Maven repository. So I tried to use [this repository](http://jaspersoft.artifactoryonline.com/jaspersoft/webapp/browserepo.html?42). But there is a conflict with version 5.5.0. So I tried to use version 5.6.0, but this also requires version 5.6.0 of JasperReports. But then mfp will not compile anymore...

In short: Could we not add it as dependency for now? Maybe you can open ticket, so that we add a dependency when we update JasperReports?

**Comment by asaunier** [comment]
> In short: Could we not add it as dependency for now?

No problem! Since the other JAR files cannot be integrated in mfp, I need to get them by hand anyway.

By the way it seems a pain in the neck to find the download locations for those JAR files. @sbrunner should we add them to the GMF scaffold at https://github.com/camptocamp/c2cgeoportal/tree/master/c2cgeoportal/scaffolds/create/print/WEB-INF/
?

**Comment by sbrunner** [comment]
No because we download them directly:
https://github.com/camptocamp/c2cgeoportal/blob/master/c2cgeoportal/scaffolds/update/CONST_Makefile_tmpl#L603-L617

I'd like to do the same with Mapfish Print but with the actually release procedure it's not possible ...
